libtenzir/builtins/operators/partitions.cpp

```diff
@@ -218,6 +220,19 @@ class plugin final : public virtual operator_plugin<partitions_operator> {
     return std::make_unique<partitions_operator>(
       std::move(expr->inner), experimental_include_ranges.has_value());
   }
+
+  auto make(invocation inv, session ctx) const
+    -> failure_or<operator_ptr> override {
+    auto expr = ast::expression{};
+    auto experimental_include_ranges = std::optional<location>{};
+    TRY(argument_parser2::operator_("partitions")
+          .add(expr, "<expr>")
+          .add("experimental_include_ranges", experimental_include_ranges)
+          .parse(inv, ctx));
+    auto [legacy, _] = split_legacy_expression(expr);
+    return std::make_unique<partitions_operator>(
+      std::move(legacy), experimental_include_ranges.has_value());
+  }
```

Comment:
@jachris this is pretty useful for testing the predicate pushdown of TQL2's `export` operator. For example, I figured out that `partitions @internal` returns _all_ partitions, including those not marked internal.